### PR TITLE
ILogger for Signal_Windows

### PR DESCRIPTION
**Pull request by Trolldemorted**
Remove all occurences of Debug.WriteLine in favour of the ILogger interface, removed unused classes, stopped leaking personal information to the log, save log to `\LocalCache\Signal-Windows.ui.log`.

I added a handler for unhandled exceptions, even #1 is properly tracked:
```
[UI] [Error] [Signal_Windows.App] UnhandledException occured in (null)/0: Exception from HRESULT: 0x80072EFE
   at Windows.Networking.Sockets.MessageWebSocketMessageReceivedEventArgs.GetDataReader()
   at System.Net.WebSockets.WinRTWebSocket.OnMessageReceived(MessageWebSocket sender, MessageWebSocketMessageReceivedEventArgs args)
```

So we now can ask for debug logs if a user reports an error.

cc @golf1052, @SERVCUBED 

Note: VS 2017 15.4.2 is out!

### CODE REVIEW

File: Signal-Windows/Logging/SignalLogging.cs [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+{
+    class SignalLogging
+    {
+        public static void SetupLogging(bool ui)
+        {
+            if (ui)
+            {
+                var console = new SignalConsoleLoggerProvider();
+                var file = new SignalFileLoggerProvider(ApplicationData.Current.LocalCacheFolder.Path + @"\Signal-Windows.ui.log", "UI");
+                LibsignalLogging.LoggerFactory.AddProvider(console);
```
Comment (SERVCUBED) [inline Signal-Windows/Logging/SignalLogging.cs]: Looks good! One thing - would it be worth putting the initialisation of the debug console logger inside an `#IF DEBUG` so it's not enabled in release mode, or does the Mobile Centre SDK (or whichever telemetry we end up using) read the debug console output too?

File: Signal-Windows/ViewModels/AddContactPageViewModel.cs [hunk 2]
```diff
@@ -147,7 +149,7 @@ private bool ValidNumber
                             }
                             catch (NumberParseException)
                             {
-                                Debug.WriteLine($"Couldn't parse {phone.Number}");
+                                Logger.LogWarning("RefreshContacts() could not parse number");
```
Comment (golf1052) [inline Signal-Windows/ViewModels/AddContactPageViewModel.cs]: Having the unparsed phone number is useful. Logging the number should be .LogDebug

File: Signal-Windows/Logging/SignalLogging.cs (hunk 1 above)
Comment (golf1052) [reply Signal-Windows/Logging/SignalLogging.cs]: Agree

**Comment by golf1052** [comment]
🚢

File: Signal-Windows/ViewModels/AddContactPageViewModel.cs (hunk 2 above)
Comment (Trolldemorted) [reply Signal-Windows/ViewModels/AddContactPageViewModel.cs]: As long as we have no way to censor phone numbers i feel very reluctant to add them to a log file with will be uploaded to the internet w/o further protection. We should censor them like Signal-Android does, I guess

**Pull request merged**